File: src/java/src/org/quickfix/fix40/Advertisement.java
package org.quickfix.fix40; 
import org.quickfix.FieldNotFound; 
import org.quickfix.Group; 
import org.quickfix.field.*; 

public class Advertisement extends Message 
{ 

  public Advertisement() 
  { 
    getHeader().setField(new MsgType("7")); 
  } 
  public Advertisement(    
    org.quickfix.field.AdvId aAdvId,    
    org.quickfix.field.AdvTransType aAdvTransType,    
    org.quickfix.field.Symbol aSymbol,    
    org.quickfix.field.AdvSide aAdvSide,    
    org.quickfix.field.Shares aShares ) 
  {  
    getHeader().setField(new MsgType("7")); 
    set(aAdvId); 
    set(aAdvTransType); 
    set(aSymbol); 
    set(aAdvSide); 
    set(aShares);  
  } 

  public void set(org.quickfix.field.AdvId value) 
  { setField(value); } 
  public org.quickfix.field.AdvId get(org.quickfix.field.AdvId value) throws FieldNotFound 
  { getField(value); return value; } 
  public org.quickfix.field.AdvId getAdvId() throws FieldNotFound 
  { org.quickfix.field.AdvId value = new org.quickfix.field.AdvId(); 
    getField(value); return value; } 
  public boolean isSet(org.quickfix.field.AdvId field) 
  { return isSetField(field); } 
  public boolean isSetAdvId() 
  { return isSetField(2); } 

  public void set(org.quickfix.field.AdvTransType value) 
  { setField(value); } 
  public org.quickfix.field.AdvTransType get(org.quickfix.field.AdvTransType value) throws FieldNotFound 
  { getField(value); return value; } 
  public org.quickfix.field.AdvTransType getAdvTransType() throws FieldNotFound 
  { org.quickfix.field.AdvTransType value = new org.quickfix.field.AdvTransType(); 
    getField(value); return value; } 
  public boolean isSet(org.quickfix.field.AdvTransType field) 
  { return isSetField(field); } 
  public boolean isSetAdvTransType() 
  { return isSetField(5); } 

  public void set(org.quickfix.field.AdvRefID value) 
  { setField(value); } 
  public org.quickfix.field.AdvRefID get(org.quickfix.field.AdvRefID value) throws FieldNotFound 
  { getField(value); return value; } 
  public org.quickfix.field.AdvRefID getAdvRefID() throws FieldNotFound 
  { org.quickfix.field.AdvRefID value = new org.quickfix.field.AdvRefID(); 
    getField(value); return value; } 
  public boolean isSet(org.quickfix.field.AdvRefID field) 
  { return isSetField(field); } 
  public boolean isSetAdvRefID() 
  { return isSetField(3); } 

  public void set(org.quickfix.field.Symbol value) 
  { setField(value); } 
  public org.quickfix.field.Symbol get(org.quickfix.field.Symbol value) throws FieldNotFound 
  { getField(value); return value; } 
  public org.quickfix.field.Symbol getSymbol() throws FieldNotFound 
  { org.quickfix.field.Symbol value = new org.quickfix.field.Symbol(); 
    getField(value); return value; } 
  public boolean isSet(org.quickfix.field.Symbol field) 
  { return isSetField(field); } 
  public boolean isSetSymbol() 
  { return isSetField(55); } 

  public void set(org.quickfix.field.SymbolSfx value) 
  { setField(value); } 
  public org.quickfix.field.SymbolSfx get(org.quickfix.field.SymbolSfx value) throws FieldNotFound 
  { getField(value); return value; } 
  public org.quickfix.field.SymbolSfx getSymbolSfx() throws FieldNotFound 
  { org.quickfix.field.SymbolSfx value = new org.quickfix.field.SymbolSfx(); 
    getField(value); return value; } 
  public boolean isSet(org.quickfix.field.SymbolSfx field) 
  { return isSetField(field); } 
  public boolean isSetSymbolSfx() 
  { return isSetField(65); } 

  public void set(org.quickfix.field.SecurityID value) 
  { setField(value); } 
  public org.quickfix.field.SecurityID get(org.quickfix.field.SecurityID value) throws FieldNotFound 
  { getField(value); return value; } 
  public org.quickfix.field.SecurityID getSecurityID() throws FieldNotFound 
  { org.quickfix.field.SecurityID value = new org.quickfix.field.SecurityID(); 
    getField(value); return value; } 
  public boolean isSet(org.quickfix.field.SecurityID field) 
  { return isSetField(field); } 
  public boolean isSetSecurityID() 
  { return isSetField(48); } 

  public void set(org.quickfix.field.IDSource value) 
  { setField(value); } 
  public org.quickfix.field.IDSource get(org.quickfix.field.IDSource value) throws FieldNotFound 
  { getField(value); return value; } 
  public org.quickfix.field.IDSource getIDSource() throws FieldNotFound 
  { org.quickfix.field.IDSource value = new org.quickfix.field.IDSource(); 
    getField(value); return value; } 
  public boolean isSet(org.quickfix.field.IDSource field) 
  { return isSetField(field); } 
  public boolean isSetIDSource() 
  { return isSetField(22); } 

  public void set(org.quickfix.field.Issuer value) 
  { setField(value); } 
  public org.quickfix.field.Issuer get(org.quickfix.field.Issuer value) throws FieldNotFound 
  { getField(value); return value; } 
  public org.quickfix.field.Issuer getIssuer() throws FieldNotFound 
  { org.quickfix.field.Issuer value = new org.quickfix.field.Issuer(); 
    getField(value); return value; } 
  public boolean isSet(org.quickfix.field.Issuer field) 
  { return isSetField(field); } 
  public boolean isSetIssuer() 
  { return isSetField(106); } 

  public void set(org.quickfix.field.SecurityDesc value) 
  { setField(value); } 
  public org.quickfix.field.SecurityDesc get(org.quickfix.field.SecurityDesc value) throws FieldNotFound 
  { getField(value); return value; } 
  public org.quickfix.field.SecurityDesc getSecurityDesc() throws FieldNotFound 
  { org.quickfix.field.SecurityDesc value = new org.quickfix.field.SecurityDesc(); 
    getField(value); return value; } 
  public boolean isSet(org.quickfix.field.SecurityDesc field) 
  { return isSetField(field); } 
  public boolean isSetSecurityDesc() 
  { return isSetField(107); } 

  public void set(org.quickfix.field.AdvSide value) 
  { setField(value); } 
  public org.quickfix.field.AdvSide get(org.quickfix.field.AdvSide value) throws FieldNotFound 
  { getField(value); return value; } 
  public org.quickfix.field.AdvSide getAdvSide() throws FieldNotFound 
  { org.quickfix.field.AdvSide value = new org.quickfix.field.AdvSide(); 
    getField(value); return value; } 
  public boolean isSet(org.quickfix.field.AdvSide field) 
  { return isSetField(field); } 
  public boolean isSetAdvSide() 
  { return isSetField(4); } 

  public void set(org.quickfix.field.Shares value) 
  { setField(value); } 
  public org.quickfix.field.Shares get(org.quickfix.field.Shares value) throws FieldNotFound 
  { getField(value); return value; } 
  public org.quickfix.field.Shares getShares() throws FieldNotFound 
  { org.quickfix.field.Shares value = new org.quickfix.field.Shares(); 
    getField(value); return value; } 
  public boolean isSet(org.quickfix.field.Shares field) 
  { return isSetField(field); } 
  public boolean isSetShares() 
  { return isSetField(53); } 

  public void set(org.quickfix.field.Price value) 
  { setField(value); } 
  public org.quickfix.field.Price get(org.quickfix.field.Price value) throws FieldNotFound 
  { getField(value); return value; } 
  public org.quickfix.field.Price getPrice() throws FieldNotFound 
  { org.quickfix.field.Price value = new org.quickfix.field.Price(); 
    getField(value); return value; } 
  public boolean isSet(org.quickfix.field.Price field) 
  { return isSetField(field); } 
  public boolean isSetPrice() 
  { return isSetField(44); } 

  public void set(org.quickfix.field.Currency value) 
  { setField(value); } 
  public org.quickfix.field.Currency get(org.quickfix.field.Currency value) throws FieldNotFound 
  { getField(value); return value; } 
  public org.quickfix.field.Currency getCurrency() throws FieldNotFound 
  { org.quickfix.field.Currency value = new org.quickfix.field.Currency(); 
    getField(value); return value; } 
  public boolean isSet(org.quickfix.field.Currency field) 
  { return isSetField(field); } 
  public boolean isSetCurrency() 
  { return isSetField(15); } 

  public void set(org.quickfix.field.TransactTime value) 
  { setField(value); } 
  public org.quickfix.field.TransactTime get(org.quickfix.field.TransactTime value) throws FieldNotFound 
  { getField(value); return value; } 
  public org.quickfix.field.TransactTime getTransactTime() throws FieldNotFound 
  { org.quickfix.field.TransactTime value = new org.quickfix.field.TransactTime(); 
    getField(value); return value; } 
  public boolean isSet(org.quickfix.field.TransactTime field) 
  { return isSetField(field); } 
  public boolean isSetTransactTime() 
  { return isSetField(60); } 

  public void set(org.quickfix.field.Text value) 
  { setField(value); } 
  public org.quickfix.field.Text get(org.quickfix.field.Text value) throws FieldNotFound 
  { getField(value); return value; } 
  public org.quickfix.field.Text getText() throws FieldNotFound 
  { org.quickfix.field.Text value = new org.quickfix.field.Text(); 
    getField(value); return value; } 
  public boolean isSet(org.quickfix.field.Text field) 
  { return isSetField(field); } 
  public boolean isSetText() 
  { return isSetField(58); } 
} 

File: src/.NET/fix40/Advertisement.cs
namespace QuickFix40
{

  public class Advertisement : Message
  {
    public Advertisement() : base(MsgType()) {}
    static QuickFix.MsgType MsgType() { return new QuickFix.MsgType("7"); }

    public Advertisement(
      QuickFix.AdvId aAdvId,
      QuickFix.AdvTransType aAdvTransType,
      QuickFix.Symbol aSymbol,
      QuickFix.AdvSide aAdvSide,
      QuickFix.Shares aShares )
    : base(MsgType()) {
      set(aAdvId);
      set(aAdvTransType);
      set(aSymbol);
      set(aAdvSide);
      set(aShares);
    }

    public void set(QuickFix.AdvId value)
    { setField(value); }
    public QuickFix.AdvId get(QuickFix.AdvId  value)
    { getField(value); return value; }
    public QuickFix.AdvId getAdvId()
    { QuickFix.AdvId value = new QuickFix.AdvId();
      getField(value); return value; }
    public bool isSet(QuickFix.AdvId field)
    { return isSetField(field); }
    public bool isSetAdvId()
    { return isSetField(2); }

    public void set(QuickFix.AdvTransType value)
    { setField(value); }
    public QuickFix.AdvTransType get(QuickFix.AdvTransType  value)
    { getField(value); return value; }
    public QuickFix.AdvTransType getAdvTransType()
    { QuickFix.AdvTransType value = new QuickFix.AdvTransType();
      getField(value); return value; }
    public bool isSet(QuickFix.AdvTransType field)
    { return isSetField(field); }
    public bool isSetAdvTransType()
    { return isSetField(5); }

    public void set(QuickFix.AdvRefID value)
    { setField(value); }
    public QuickFix.AdvRefID get(QuickFix.AdvRefID  value)
    { getField(value); return value; }
    public QuickFix.AdvRefID getAdvRefID()
    { QuickFix.AdvRefID value = new QuickFix.AdvRefID();
      getField(value); return value; }
    public bool isSet(QuickFix.AdvRefID field)
    { return isSetField(field); }
    public bool isSetAdvRefID()
    { return isSetField(3); }

    public void set(QuickFix.Symbol value)
    { setField(value); }
    public QuickFix.Symbol get(QuickFix.Symbol  value)
    { getField(value); return value; }
    public QuickFix.Symbol getSymbol()
    { QuickFix.Symbol value = new QuickFix.Symbol();
      getField(value); return value; }
    public bool isSet(QuickFix.Symbol field)
    { return isSetField(field); }
    public bool isSetSymbol()
    { return isSetField(55); }

    public void set(QuickFix.SymbolSfx value)
    { setField(value); }
    public QuickFix.SymbolSfx get(QuickFix.SymbolSfx  value)
    { getField(value); return value; }
    public QuickFix.SymbolSfx getSymbolSfx()
    { QuickFix.SymbolSfx value = new QuickFix.SymbolSfx();
      getField(value); return value; }
    public bool isSet(QuickFix.SymbolSfx field)
    { return isSetField(field); }
    public bool isSetSymbolSfx()
    { return isSetField(65); }

    public void set(QuickFix.SecurityID value)
    { setField(value); }
    public QuickFix.SecurityID get(QuickFix.SecurityID  value)
    { getField(value); return value; }
    public QuickFix.SecurityID getSecurityID()
    { QuickFix.SecurityID value = new QuickFix.SecurityID();
      getField(value); return value; }
    public bool isSet(QuickFix.SecurityID field)
    { return isSetField(field); }
    public bool isSetSecurityID()
    { return isSetField(48); }

    public void set(QuickFix.IDSource value)
    { setField(value); }
    public QuickFix.IDSource get(QuickFix.IDSource  value)
    { getField(value); return value; }
    public QuickFix.IDSource getIDSource()
    { QuickFix.IDSource value = new QuickFix.IDSource();
      getField(value); return value; }
    public bool isSet(QuickFix.IDSource field)
    { return isSetField(field); }
    public bool isSetIDSource()
    { return isSetField(22); }

    public void set(QuickFix.Issuer value)
    { setField(value); }
    public QuickFix.Issuer get(QuickFix.Issuer  value)
    { getField(value); return value; }
    public QuickFix.Issuer getIssuer()
    { QuickFix.Issuer value = new QuickFix.Issuer();
      getField(value); return value; }
    public bool isSet(QuickFix.Issuer field)
    { return isSetField(field); }
    public bool isSetIssuer()
    { return isSetField(106); }

    public void set(QuickFix.SecurityDesc value)
    { setField(value); }
    public QuickFix.SecurityDesc get(QuickFix.SecurityDesc  value)
    { getField(value); return value; }
    public QuickFix.SecurityDesc getSecurityDesc()
    { QuickFix.SecurityDesc value = new QuickFix.SecurityDesc();
      getField(value); return value; }
    public bool isSet(QuickFix.SecurityDesc field)
    { return isSetField(field); }
    public bool isSetSecurityDesc()
    { return isSetField(107); }

    public void set(QuickFix.AdvSide value)
    { setField(value); }
    public QuickFix.AdvSide get(QuickFix.AdvSide  value)
    { getField(value); return value; }
    public QuickFix.AdvSide getAdvSide()
    { QuickFix.AdvSide value = new QuickFix.AdvSide();
      getField(value); return value; }
    public bool isSet(QuickFix.AdvSide field)
    { return isSetField(field); }
    public bool isSetAdvSide()
    { return isSetField(4); }

    public void set(QuickFix.Shares value)
    { setField(value); }
    public QuickFix.Shares get(QuickFix.Shares  value)
    { getField(value); return value; }
    public QuickFix.Shares getShares()
    { QuickFix.Shares value = new QuickFix.Shares();
      getField(value); return value; }
    public bool isSet(QuickFix.Shares field)
    { return isSetField(field); }
    public bool isSetShares()
    { return isSetField(53); }

    public void set(QuickFix.Price value)
    { setField(value); }
    public QuickFix.Price get(QuickFix.Price  value)
    { getField(value); return value; }
    public QuickFix.Price getPrice()
    { QuickFix.Price value = new QuickFix.Price();
      getField(value); return value; }
    public bool isSet(QuickFix.Price field)
    { return isSetField(field); }
    public bool isSetPrice()
    { return isSetField(44); }

    public void set(QuickFix.Currency value)
    { setField(value); }
    public QuickFix.Currency get(QuickFix.Currency  value)
    { getField(value); return value; }
    public QuickFix.Currency getCurrency()
    { QuickFix.Currency value = new QuickFix.Currency();
      getField(value); return value; }
    public bool isSet(QuickFix.Currency field)
    { return isSetField(field); }
    public bool isSetCurrency()
    { return isSetField(15); }

    public void set(QuickFix.TransactTime value)
    { setField(value); }
    public QuickFix.TransactTime get(QuickFix.TransactTime  value)
    { getField(value); return value; }
    public QuickFix.TransactTime getTransactTime()
    { QuickFix.TransactTime value = new QuickFix.TransactTime();
      getField(value); return value; }
    public bool isSet(QuickFix.TransactTime field)
    { return isSetField(field); }
    public bool isSetTransactTime()
    { return isSetField(60); }

    public void set(QuickFix.Text value)
    { setField(value); }
    public QuickFix.Text get(QuickFix.Text  value)
    { getField(value); return value; }
    public QuickFix.Text getText()
    { QuickFix.Text value = new QuickFix.Text();
      getField(value); return value; }
    public bool isSet(QuickFix.Text field)
    { return isSetField(field); }
    public bool isSetText()
    { return isSetField(58); }

  };

}

